src/v4/base/BaseHook.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IPreExecutionHook} from "../interfaces/IHook.sol";
+import {ResolvedOrder} from "./ReactorStructs.sol";
+import {IPermit2} from "permit2/src/interfaces/IPermit2.sol";
+import {Permit2Lib} from "../lib/Permit2Lib.sol";
+import {IAuctionResolver} from "../interfaces/IAuctionResolver.sol";
+
+/// @notice Base hook contract that enforces token transfer logic
+/// @dev inherit from this contract for default token transfer logic using permit2's Signature transfer
+abstract contract BasePreExecutionHook is IPreExecutionHook {
+    using Permit2Lib for ResolvedOrder;
+
+    /// @notice Permit2 instance for signature verification and token transfers
+    IPermit2 public immutable PERMIT2;
+
+    constructor(IPermit2 _permit2) {
+        PERMIT2 = _permit2;
+    }
+
+    /// @inheritdoc IPreExecutionHook
+    function preExecutionHook(address filler, ResolvedOrder calldata resolvedOrder) external override {
```

Comment:
Should this only be callable by the Reactor?

Reply:
not sure I follow this comment 
>Since the Reactor uses nonReentrant, there's no need to add it here, right?

But I think an `onlyReactor` modifier makes sense?